### PR TITLE
Node e2e: collect logs if the test fails unexpectedly

### PR DESCRIPTION
**Pull request by yujuhong**
This only works for nodes with journald.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36181)

### CODE REVIEW

File: test/e2e_node/remote/remote.go [hunk 1]
```diff
@@ -294,6 +296,32 @@ func RunRemote(archive string, host string, cleanup bool, junitFilePrefix string
 		aggErrs = append(aggErrs, err)
 	}
 
+	if err != nil {
+		// Encountered an unexpected error. The remote test harness may not
+		// have finished retrieved and stored all the logs in this case. Try
+		// to some logs for debugging purpose.
```
Comment (Random-Liu) [inline test/e2e_node/remote/remote.go]: try to get

**Comment by Random-Liu** [comment]
LGTM with a nit.

**Comment by yujuhong** [comment]
Addressed the nit. Also manually verified the PR by triggering a node reboot (and/or killing the ginkgo process) during the node e2e tests, and saw the logs get collected back on to my desktop.

**Comment by yujuhong** [comment]
@k8s-bot cvm gke e2e test this

**Comment by yujuhong** [comment]
@k8s-bot node e2e test this

**Pull request merged**